### DISCUSSION TITLE
How to disable river ui logging?

### DISCUSSION THREAD

**Discussion by TArch64** [riverqueue/riverui#161] [Q&A] [answered]
Hi guys. Is there any way to disable logging each request? As a developer the only logs that I need that river ui successfully started but currently it spamming my console with access logs. Now it's quite disturbing because it makes harder to find my app logs

<img width="1800" alt="image" src="https://github.com/user-attachments/assets/a5a0572a-78ca-4ecb-8c2d-71a8d353470d">

**Comment by TArch64** [comment]
I have `RIVER_DEBUG=0` in my envs but it doesnt off this logs. Also I searched the code and doesnt find any env to set log level

**Comment by bgentry** [comment]
You’re running it as a handler within another Go app, correct? You can pass the server opts a logger with a warn or error level and you shouldn’t see much of anything emitted https://pkg.go.dev/riverqueue.com/riverui#ServerOpts

**Reply by TArch64** [reply]
No, I’m running it using docker image. I built it manually to pass another base url for frontend

**Reply by TArch64** [reply]
I dont like idea to have river ui as a part of my backend. So I would prefer to have it in container

**Reply by bgentry** [reply]
Got it, there's no config for this for the `riverui` executable as of today. I think it would make sense to have a `RIVER_LOG_LEVEL` env for the main executable (`cmd/riverui/main.go`) that accepts any of the [common names of log levels'](https://pkg.go.dev/log/slog#hdr-Levels) strings (debug, info, notice, warning, error, etc) along with an additional value `"none"` to disable it altogether.

If you want to put together a small PR to do that it should be pretty quick. I'd be happy to review and ship it after that :v:

**Reply by TArch64** [reply]
@bgentry Opened a pull request https://github.com/riverqueue/riverui/pull/183. Decided not to add a `none` log level because it's good to have error or warn logs to know if there any issues in configuration even in production environment.

Setting warn log level works good for me

**Comment by TArch64** [comment, marked as answer]
Thanks for your help! Added `RIVER_LOG_LEVEL` [env](https://github.com/riverqueue/riverui?tab=readme-ov-file#logging-configuration) which solves all my issues so closing the discussion
